src/main/java/com/example/weatherwaveapi/config/ApplicationConfiguration.java

```diff
@@ -9,17 +10,17 @@
 import java.time.Duration;
 
 @Configuration
+@RequiredArgsConstructor
+@EnableConfigurationProperties(GeneralSettings.class)
 public class ApplicationConfiguration {
```

Comment:
I suggested not changing this file. This class already contains the Configuration annotation, so it can pull the settings from the Yaml file

Reply:
We need EnableConfigurationProperties(GeneralSettings.class) annotation here to unable creating setting classes with @ConfigurationProperties(...)

Reply:
the fewer dependencies the better, why do we need to use an additional class if we can directly take the data we need from the Yaml file?

Reply:
Hm, do you mean to work without setting classes and use properties directly by @Value? If yes I am not sure this is a good idea, because with using a special class for settings we have a more OOP friendly way to work with our properties as with an objects + it will be more organized, so we can just create an instance of settings class and use it. It also will be more adoptability, because in case of refactoring you only need to update the settings class rather than hunting down every reference to individual properties scattered throughout our codebase